### ISSUE TITLE
check_ping: CRITICAL - Could not interpret output from ping command

### ISSUE THREAD

**Issue by waja** [bodsch/docker-icinga2#46]
After Starting the container and montoring, check_ping (and probably others, as check_icmp and check_dhcp) is not working.

```
sh-4.4# su - nagios
$ /bin/ping -n -U -w 30 -c 5 127.0.0.1
ping: socket: Address family not supported by protocol
$ /bin/ping -4 -n -U -w 30 -c 5 127.0.0.1
ping: socket: Operation not permitted
sh-4.4# setcap "cap_net_raw+ep" /bin/ping
Failed to set capabilities on file `/bin/ping' (Operation not supported)
The value of the capability argument is not permitted for a file. Or the file is not a regular (non-symlink) file
sh-4.4# getcap /bin/ping
Failed to get capabilities of file `/bin/ping' (Operation not supported)
sh-4.4# capsh --print
Current: = cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap+eip
Bounding set =cap_chown,cap_dac_override,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_net_bind_service,cap_net_raw,cap_sys_chroot,cap_mknod,cap_audit_write,cap_setfcap
Securebits: 00/0x0/1'b0
 secure-noroot: no (unlocked)
 secure-no-suid-fixup: no (unlocked)
 secure-keep-caps: no (unlocked)
uid=0(root)
gid=0(root)
groups=
```

**Comment by waja** [comment]
A quick fix:
```
sh-4.4# chmod +s /bin/ping
sh-4.4# su - nagios
$ /bin/ping -4 -n -U -w 30 -c 5 127.0.0.1
PING 127.0.0.1 (127.0.0.1) 56(84) bytes of data.
64 bytes from 127.0.0.1: icmp_seq=1 ttl=64 time=0.097 ms
64 bytes from 127.0.0.1: icmp_seq=2 ttl=64 time=0.053 ms
^C
--- 127.0.0.1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1000ms
rtt min/avg/max/mdev = 0.053/0.075/0.097/0.022 ms
```

Maybe you just want to do that in your entrypoint script?

**Comment by waja** [comment]
Oh ... maybe installing `iputils` might also fix it, see https://github.com/gliderlabs/docker-alpine/issues/253#issuecomment-299518713

**Comment by waja** [comment]
> Oh ... maybe installing iputils

forget about that, you are using debian and there is `iputils` already installed.

**Comment by bodsch** [comment]
Danke fürs Issue! <3

**Comment by waja** [comment]
You [fix](https://github.com/bodsch/docker-icinga2/commit/988b698ce5f51629978da91ecb6aa823f65219b8#diff-d0fe5bf5d174488c9c5e50a3afeb67ab) results into:

```
icinga2-master    | [2019-02-09 12:01:11]  prepare system
icinga2-master    | 
icinga2-master    |  - setting cap_net_raw+ep for some check scripts
icinga2-master    | 
icinga2-master    | Failed to set capabilities on file `/usr/lib/nagios/plugins/check_icmp' (Operation not supported)
icinga2-master    | The value of the capability argument is not permitted for a file. Or the file is not a regular (non-symlink) file
icinga2-master    | setcap for check_icmp and check_dhcp failed.
icinga2-master    | Please refer README.Debian.gz for using plugins needing
icinga2-master    | higher privileges!
```

Which does not solve the problem, as setcap has issues in container environments. `chmod +s /bin/ping` did solve the issue for me at least.

**Comment by waja** [comment]
```
sh-4.4# su - nagios
$ /usr/lib/nagios/plugins/check_icmp 127.0.0.1
check_icmp: Failed to obtain ICMP socket: Operation not permitted
$ logout
sh-4.4# chmod +s /usr/lib/nagios/plugins/check_icmp
sh-4.4# su - nagios
$ /usr/lib/nagios/plugins/check_icmp 127.0.0.1
OK - 127.0.0.1: rta 0.197ms, lost 0%|rta=0.197ms;200.000;500.000;0; pl=0%;40;80;; rtmax=0.941ms;;;; rtmin=0.007ms;;;; 
$ /usr/lib/nagios/plugins/check_dhcp          
Error: Could not bind socket to interface eth0.  Check your privileges...
$ logout
sh-4.4# chmod +s /usr/lib/nagios/plugins/check_dhcp
sh-4.4# su - nagios
$ /usr/lib/nagios/plugins/check_dhcp 
^C
```
I would suggest to do the following (maybe quik&dirty):
```
chmod +x /bin/ping /usr/lib/nagios/plugins/check_icmp /usr/lib/nagios/plugins/check_dhcp
```

**Comment by bodsch** [comment]
Stange!
I use the official monitoring-plugins for the debian distribution.
They should already set the setcap at installation time.

I take a deeper look

**Comment by bodsch** [comment]
the result on my local machine, fresh build of the latest icnaga2-base:

```
13:30 $ make base-shell
root@bodsch-icinga2:/etc/icinga2#  /bin/ping -n -U -w 30 -c 5 127.0.0.1
PING 127.0.0.1 (127.0.0.1) 56(84) bytes of data.
64 bytes from 127.0.0.1: icmp_seq=1 ttl=64 time=0.069 ms
64 bytes from 127.0.0.1: icmp_seq=2 ttl=64 time=0.084 ms
^C
--- 127.0.0.1 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1021ms
rtt min/avg/max/mdev = 0.069/0.076/0.084/0.011 ms
root@bodsch-icinga2:/etc/icinga2#
```
```
root@bodsch-icinga2:/etc/icinga2# /usr/lib/nagios/plugins/check_icmp 127.0.0.1
OK - 127.0.0.1: rta 0.022ms, lost 0%|rta=0.022ms;200.000;500.000;0; pl=0%;40;80;; rtmax=0.052ms;;;; rtmin=0.014ms;;;;
root@bodsch-icinga2:/etc/icinga2#
```
But ... the docker run command includes `--cap-add=SYS_ADMIN`

**Comment by waja** [comment]
Okay ... I adjusted `docker-compose.yml`:
```
  icinga2-master:                                                               
    cap_add:                                                                    
      - SYS_ADMIN 
```
```
# docker ps | grep master
0a7adfca8b95        bodsch/docker-icinga2:2.10.2-master   "/init/run.sh"      27 minutes ago      Up 23 minutes (healthy)   0.0.0.0:5665->5665/tcp, 0.0.0.0:8080->8080/tcp   icinga2-master
# docker inspect 0a7adfca8b95 | grep -2 CapAdd | tail -3
            "CapAdd": [
                "SYS_ADMIN"
            ],
```
```
sh-4.4# su - nagios
$ /bin/ping -n -U -w 30 -c 5 127.0.0.1
ping: socket: Address family not supported by protocol
$ /usr/lib/nagios/plugins/check_icmp 127.0.0.1
check_icmp: Failed to obtain ICMP socket: Operation not permitted
```
Anyway ... you tried this in https://github.com/bodsch/docker-icinga2/issues/46#issuecomment-462040974 as user `root` and NOT `nagios`.

When you want to try this as user `nagios` `sed -i "s#nagios:/bin/false#nagios:/bin/sh#" /etc/passwd` might become handy. :)

**Comment by waja** [comment]
Anyway ... even with enabled all capabilities:
```
# docker ps | grep master
5b662b02ef1c        bodsch/docker-icinga2:2.10.2-master   "/init/run.sh"      6 seconds ago       Up 5 seconds (health: starting)   0.0.0.0:5665->5665/tcp, 0.0.0.0:8080->8080/tcp   icinga2-master
# docker inspect 5b662b02ef1c | grep -2 CapAdd | tail -3
            "CapAdd": [
                "ALL"
            ],
# docker-compose exec icinga2-master sh
sh-4.4# sed -i "s#nagios:/bin/false#nagios:/bin/sh#" /etc/passwd
sh-4.4# su - nagios
$ /usr/lib/nagios/plugins/check_icmp 127.0.0.1
check_icmp: Failed to obtain ICMP socket: Operation not permitted
```

**Comment by waja** [comment]
Okay ... I checked you latest container (45daccca00b2):

```
icinga2-master    | [2019-02-09 17:35:18]  setting cap_net_raw+ep for some check scripts
icinga2-master    | Failed to set capabilities on file `/usr/lib/nagios/plugins/check_icmp' (Operation not supported)
icinga2-master    | The value of the capability argument is not permitted for a file. Or the file is not a regular (non-symlink) file
icinga2-master    | [2019-02-09 17:35:18] ERROR setcap for check_icmp and check_dhcp failed.
icinga2-master    | [2019-02-09 17:35:18] ERROR Please refer README.Debian.gz for using plugins needing
icinga2-master    | [2019-02-09 17:35:18] ERROR higher privileges!
```

```
# docker-compose exec icinga2-master sh
WARNING: The ICINGAWEB_DIRECTOR variable is not set. Defaulting to a blank string.
sh-4.4# sed -i "s#nagios:/bin/false#nagios:/bin/sh#" /etc/passwd
sh-4.4# su - nagios
$ ls -la /bin/ping /usr/lib/nagios/plugins/check_icmp /usr/lib/nagios/plugins/check_dhcp 
-rwxr-xr-x 1 root root 61240 Nov 10  2016 /bin/ping
-rwxr-xr-x 1 root root 59952 Jan 19  2017 /usr/lib/nagios/plugins/check_dhcp
-rwxr-xr-x 1 root root 63728 Jan 19  2017 /usr/lib/nagios/plugins/check_icmp
$ /bin/ping -n -U -w 30 -c 5 127.0.0.1
ping: socket: Address family not supported by protocol
$ /usr/lib/nagios/plugins/check_icmp 127.0.0.1
check_icmp: Failed to obtain ICMP socket: Operation not permitted
```

This should be fixed by #49
